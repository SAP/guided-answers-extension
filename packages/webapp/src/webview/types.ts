import type {
    GuidedAnswerNode,
    GuidedAnswerTree,
    GuidedAnswerTreeSearchResult
} from '@sap/guided-answers-extension-types';

export interface AppState {
    loading: boolean;
    query: string;
    guidedAnswerTreeSearchResult: GuidedAnswerTreeSearchResult;
    activeGuidedAnswerNode: GuidedAnswerNode[];
    activeGuidedAnswer?: GuidedAnswerTree;
    betaFeatures: boolean;
    searchResultCount: number;
    guideFeedback: null | boolean;
    selectedProductFilters: string[];
    selectedComponentFilters: string[];
    pageSize: number;
    feedbackStatus: boolean;
    feedbackResponse: boolean;
}
