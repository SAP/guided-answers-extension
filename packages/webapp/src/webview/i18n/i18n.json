{
    "GUIDED_ANSWER_HEADER": "Guide",
    "SEARCH_GUIDED_ANSWERS": "Search Guided Answers",
    "ALL_ANSWERS": "Home",
    "STEP_BACK": "Step back",
    "RESTART": "Restart",
    "GUIDED_ANSWERS": "Guided Answers",
    "NO_ANSWERS_FOUND": "No answers found",
    "PLEASE_MODIFY_SEARCH": "Please modfiy search",
    "THIS_SOLVED_MY_ISSUE": "This solved my issue",
    "THIS_DID_NOT_SOLVED_MY_ISSUE": "This did not solve my issue",
    "THANK_YOU_TEXT": "We are glad to hear that your issue has been resolved and we hope you enjoy using the Guided Answers application.",
    "THANKS": "Thanks!",
    "HOME": "Home",
    "SEND": "Send",
    "CLOSE": "Close",
    "ISSUE_IS_NOT_RESOLVED": "Issue is not resolved",
    "WE_ARE_SORRY_TO_HEAR_THAT_YOUR_ISSUE_IS_NOT_YET_RESOLVED": "We are sorry to hear that your issue is not yet resolved.",
    "PLEASE_TELL_US_IF_THIS_ANSWER_WAS_HELPFUL": "Please tell us if this answer was helpful",
    "THERE_ARE_SEVERAL_OPTIONS_FOR_GETTING_FURTHER_ASSISTANCE": "There are several options for getting further assistance:",
    "SEARCH_FOR_ANOTHER_GUIDED_ANSWER": "Search for another Guided Answer",
    "START_AN_EXPERT_CHAT": "Start an Expert Chat",
    "SCHUEDULE_AN_EXPERT": "Schedule an Expert",
    "OPEN_AN_INCIDENT": "Open an Incident",
    "ASK_THE_SAP_COMMUNITY": "Ask the SAP Community",
    "FEEDBACK": "Feedback",
    "FEEDBACK_DIALOG_TITLE": "Is this content helpful?",
    "FEEDBACK_DIALOG_SUBTEXT": "If you have suggestions on how to improve this content we would love to hear them!",
    "FEEDBACK_DIALOG_DISCLAIMER": "Your feedback is anonymous, we do not collect any personal data.",
    "FEEDBACK_DIALOG_SUGGESTION": "Your suggestion",
    "FEEDBACK_SENT_DIALOG_TITLE": "Message sent",
    "FEEDBACK_SENT_DIALOG_SUBTEXT": "Thanks for the feedback!"
}
