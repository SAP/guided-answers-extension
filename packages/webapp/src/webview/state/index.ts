export { actions, store } from './store';
export { getInitialState } from './reducers';
