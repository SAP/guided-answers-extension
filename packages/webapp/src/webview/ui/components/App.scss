html {
    height: 100%;
}
* {
    font-family: var(--vscode-font-family);
    -webkit-font-smoothing: antialiased;
}

h1 {
    font-weight: 300;
    color: var(--vscode-foreground);
    font-size: 24px;
}

h2 {
    font-size: 13px;
}

.guided-answer {
    &__header {
        overflow: hidden;
        position: fixed; /* Set the navbar to fixed position */
        top: 0; /* Position the navbar at the top of the page */
        width: 100%;
        z-index: 10;
        background-color: var(--vscode-editor-background);
        &__title {
            font-weight: 600;
            margin: 0px;
            color: var(--vscode-settings-headerForeground);
            line-height: 29px;
        }
        &__subtitle {
            color: var(--vscode-editorCodeLens-foreground, #999);
            font-size: 2em;
        }
        &__goback {
            height: 26px;
            margin-top: 17px;
        }
        &__logoAndTitle {
            display: flex;
            align-items: center;
            margin-top: 40px;
        }
        &__searchField {
            display: flex;
            align-items: center;
            flex-direction: column;
            margin-top: -28px;
            margin-bottom: 40px;
        }
    }
    &__container {
        margin-top: 111px;
        padding-bottom: 10px;
    }
    &__tree {
        width: 100%;
        height: auto;
        cursor: pointer;
        text-align: left;
        display: flex;
        border: none;
        &:hover {
            background-color: var(--vscode-list-hoverBackground);
        }

        &__ul {
            padding: 15px;
            &h3 {
                margin-bottom: 10px;
            }
        }
        &__title {
            font-size: 14px;
            color: var(--vscode-settings-headerForeground);
            margin-bottom: 10px;
            margin-top: 0px;
        }
        &__desc {
            color: var(--vscode-foreground);
            height: auto;
            font-size: 13px;
        }
    }
}

.tree-search-field {
    width: 33%;
    min-width: 250px;
}

.tree-search-icon {
    margin-top: 5px;
    text-align: center;
}

.tree-item {
    margin: 0;
}

#sap-logo {
    width: 59px;
}

.striped-list {
    padding-left: 0px;
    margin-top: 0px;
}

ul.striped-list > li:nth-of-type(odd) {
    background-color: var(--vscode-breadcrumbPicker-background);
    padding-left: 0px;
}

li {
    background-color: var(--vscode-editor-background);
    list-style-type: none;
}
