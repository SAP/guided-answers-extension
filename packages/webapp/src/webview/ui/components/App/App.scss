html {
    height: 100%;
}
* {
    font-family: var(--vscode-font-family);
    -webkit-font-smoothing: antialiased;
}

h1 {
    color: var(--vscode-settings-headerForeground);
    font-weight: 700;
    font-size: 24px;
    margin-top: 5px;
    margin-bottom: 5px;
}

.bottom-section {
    display: flex;
    flex-direction: column;
}

.component-and-product-container {
    margin-top: 10px;
    display: flex;
    flex-direction: row;
    color: var(--vscode-foreground);
    font-size: 11px;
    font-weight: 700;
    margin-right: 15px;
}

.bottom-title {
    font-weight: 400;
}

.guided-answer {
    display: flex;
    width: 100%;
    flex-direction: column;
    &__container {
        width: 100%;
        height: 90vh;
    }
    &__tree {
        width: 100%;
        height: auto;
        cursor: pointer;
        text-align: left;
        display: flex;
        border: none;
        &:hover {
            background-color: var(--vscode-list-hoverBackground);
            outline: 1px solid var(--vscode-focusBorder, #007fd4);
            outline-offset: -1px;
        }
        button:focus-visible {
            outline: 1px solid var(--vscode-focusBorder, #007fd4);
        }
        &__ul {
            padding: 15px;
            &h3 {
                margin-bottom: 10px;
            }
        }
        &__title {
            font-size: 14px;
            color: var(--vscode-settings-headerForeground);
            margin-bottom: 10px;
            margin-top: 0px;
        }
        &__desc {
            color: var(--vscode-foreground);
            height: auto;
            font-size: 13px;
            -webkit-line-clamp: 3;
            -webkit-box-orient: vertical;
            display: -webkit-box;
            line-height: 18px;
            max-height: 54px;
            text-overflow: ellipsis;
            overflow: hidden;
            max-width: 700px;
        }
        &__product {
            margin-right: 15px;
        }
    }
}

.tree-search-field {
    width: 50%;
    min-width: 250px;
}

.tree-search-icon {
    margin-top: 5px;
    text-align: center;
}

.tree-item {
    background-color: var(--vscode-editor-background);
    list-style-type: none;
    margin: 0;
}

.striped-list {
    padding-left: 0;
    margin-top: 20px;
}

.infinite-scroll-component > li:nth-of-type(odd) {
    background-color: var(--vscode-breadcrumbPicker-background);
    padding-left: 0;
}

.bold-text {
    font-weight: 800;
}

.text-underline {
    text-decoration: underline;
}

#loading-indicator {
    width: 100%;
    margin-top: 60px;
    margin-bottom: 50px;
}
