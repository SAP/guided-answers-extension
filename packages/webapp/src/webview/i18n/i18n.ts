import { use } from 'i18next';
import { initReactI18next } from 'react-i18next';
import * as i18nEn from './i18n.json';

export { i18n } from 'i18next';

export function initI18n(): void {
    use(initReactI18next)
        .init({
            resources: {
                en: {
                    translation: i18nEn
                }
            },
            lng: 'en',
            fallbackLng: 'en',
            interpolation: {
                escapeValue: false
            }
        })
        .catch((error) => console.error(error));
}
