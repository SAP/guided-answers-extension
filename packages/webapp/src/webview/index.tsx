import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';
import { App } from './ui/components/App';
import { actions, store } from './state';
import { initI18n } from './i18n';

/**
 * Initialization of i18n, icons
 */
initI18n();

/**
 * Render app
 */
render(
    <Provider store={store}>
        <App />
    </Provider>,
    document.getElementById('root') as HTMLElement
);

/**
 * Fire action which is captured in hosting app (ide-extension) to signal that we are ready to receive messages
 */
actions.webviewReady();

/**Once webview is ready we focus to search field */
document.getElementById('search-field')?.focus();
