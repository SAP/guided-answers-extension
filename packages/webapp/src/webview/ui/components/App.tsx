import type { ReactElement } from 'react';
import React from 'react';
import { useSelector } from 'react-redux';
import { VscSearch } from 'react-icons/vsc';
import { VSCodeButton, VSCodeTextField } from '@vscode/webview-ui-toolkit/react';
import { AppState } from '../../types';
import './App.scss';
import { actions } from '../../state';
import { GuidedAnswerNode } from './GuidedAnswerNode';

import Logo from './sap-logo.svg';

let timer: NodeJS.Timeout;

/**
 * Start element for Guided Answers Extension app.
 *
 * @returns - react elements for this app
 */
export function App(): ReactElement {
    const appState = useSelector<AppState, AppState>((state) => state);

    let content;
    if (appState.activeGuidedAnswerNode.length > 0) {
        content = <GuidedAnswerNode />;
    } else if (appState.guidedAnswerTrees) {
        content = (
            <>
                <VSCodeTextField
                    className="tree-search-field"
                    value={appState.query}
                    placeholder="Search Guided Answers"
                    onInput={(e: any) => {
                        console.log(`Value changed`, e);
                        const newValue = e?.target?.value;
                        if (newValue !== undefined) {
                            clearTimeout(timer);
                            actions.setQueryValue(newValue);
                            timer = setTimeout(() => {
                                actions.searchTree(newValue);
                            }, 400);
                        }
                    }}>
                    <span slot="end" className="tree-search-icon">
                        <VscSearch />
                    </span>
                </VSCodeTextField>
                {appState.guidedAnswerTrees.map((tree, index) => {
                    return (
                        <div key={`tree-item-${index}`} className="tree-item">
                            <div
                                className="guided-answer__tree"
                                onClick={(): void => {
                                    actions.setActiveTree(tree);
                                    actions.selectNode(tree.FIRST_NODE_ID);
                                }}>
                                <ul className="guided-answer__tree__ul">
                                    <h3 className="guided-answer__tree__title">{tree.TITLE}</h3>
                                    {tree.DESCRIPTION && (
                                        <span className="guided-answer__tree__desc">{tree.DESCRIPTION}</span>
                                    )}
                                </ul>
                            </div>
                        </div>
                    );
                })}
            </>
        );
    }
    const backButton =
        appState.activeGuidedAnswerNode.length > 0 ? (
            <VSCodeButton
                className="guided-answer__header__goback"
                onClick={(): void => {
                    actions.goToPreviousPage();
                }}>
                Go back
            </VSCodeButton>
        ) : (
            <></>
        );
    return (
        <div className="guided-answer">
            <div className="guided-answer__header">
                <div className="guided-answer__header__sub">
                    <span id="sap-logo">
                        <Logo />
                    </span>
                    <h1 className="guided-answer__header__title">Guided Answers</h1>
                    <span className="guided-answer__header__subtitle">
                        {appState.activeGuidedAnswer ? ': ' + appState.activeGuidedAnswer.TITLE : ''}
                    </span>
                </div>
                {backButton}
            </div>
            <main className="guided-answer__container">{content}</main>
        </div>
    );
}
