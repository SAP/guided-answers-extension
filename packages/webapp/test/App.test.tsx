import React from 'react';
import { shallow } from 'enzyme';
import { App } from '../src/webview/ui/components/App';
import { initI18n } from '../src/webview/i18n';
import { actions } from '../src/webview/state';
import { AppState } from '../src/webview/types';

jest.mock('@vscode/webview-ui-toolkit/react', () => ({
    VSCodeTextField: () => (
        <>
            <div>SearchField</div>
        </>
    )
}));

jest.mock('../src/webview/state', () => {
    return {
        actions: {
            setActiveTree: jest.fn(),
            selectNode: jest.fn()
        }
    };
});

jest.mock('react-redux', () => {
    const mockedResult = {
        AVAILABILITY: 'PUBLIC',
        DESCRIPTION: 'This is a troubleshooting guide to solve the issues while using SAP Fiori tools',
        FIRST_NODE_ID: 45995,
        TITLE: 'SAP Fiori tools',
        TREE_ID: 3046,
        PRODUCT: 'Product A, Product B',
        COMPONENT: 'comp-a, comp-b'
    };
    return {
        ...jest.requireActual('react-redux'),
        useSelector: jest
            .fn()
            .mockReturnValueOnce({
                loading: false,
                query: '',
                guidedAnswerTreeSearchResult: {
                    trees: [],
                    resultSize: 0,
                    productFilters: [],
                    componentFilters: []
                },
                activeGuidedAnswerNode: [],
                activeGuidedAnswer: undefined
            } as unknown as AppState)
            .mockReturnValueOnce({
                activeGuidedAnswerNode: [],
                guidedAnswerTreeSearchResult: {
                    trees: [mockedResult],
                    resultSize: 1,
                    productFilters: [],
                    componentFilters: []
                },
                query: 'fiori tools'
            })
            .mockReturnValueOnce({ activeGuidedAnswerNode: [{ a: 0 }, { b: 0 }] })
            .mockReturnValueOnce({ loading: true, activeGuidedAnswerNode: [] })
    };
});

describe('<NoAnswersFound />', () => {
    let wrapper: any;
    initI18n();
    beforeEach(() => {
        wrapper = shallow(<App />);
    });

    afterEach(() => {
        jest.clearAllMocks();
        wrapper.unmount();
    });

    it('Should render a App component - No Answers', () => {
        expect(wrapper.find('.guided-answer').length).toBe(1);
        expect(wrapper.find('Header').length).toBe(1);
        expect(wrapper.find('.guided-answer__container').length).toBe(1);
        expect(wrapper.find('NoAnswersFound').length).toBe(1);
    });

    it('Should render a App component with result', () => {
        expect(wrapper.find('.guided-answer').length).toBe(1);
        expect(wrapper.find('Header').length).toBe(1);
        expect(wrapper.find('.guided-answer__container').length).toBe(1);
        expect(wrapper.find('.striped-list').length).toBe(1);
        expect(wrapper.find('.tree-item').length).toBe(1);
        expect(wrapper.find('.guided-answer__tree__title').length).toBe(1);
        expect(wrapper.find('.guided-answer__tree__ul').length).toBe(1);
        expect(wrapper.find('.guided-answer__tree__desc').text()).toBe(
            'This is a troubleshooting guide to solve the issues while using SAP Fiori tools'
        );
        expect(wrapper.find('.guided-answer__tree__product').text()).toBe('Product: Product A');
        expect(wrapper.find('.guided-answer__tree__component').text()).toBe('Component: comp-a');
        //Test click event
        wrapper.find('.guided-answer__tree').simulate('click');
        expect(actions.setActiveTree).toBeCalled();
        expect(actions.selectNode).toHaveBeenCalledWith(45995);
    });

    it('Should render a App component with GuidedAnswerNode component', () => {
        expect(wrapper.find('.guided-answer').length).toBe(1);
        expect(wrapper.find('Header').length).toBe(1);
        expect(wrapper.find('.guided-answer__container').length).toBe(1);
        expect(wrapper.find('GuidedAnswerNode').length).toBe(1);
    });

    it('Should render loading indicator', () => {
        expect(wrapper.find('#loading-indicator').length).toBe(1);
    });
});
