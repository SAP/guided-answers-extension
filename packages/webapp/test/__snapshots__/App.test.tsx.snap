// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Match snapshot of component <App/> in initial mode 1`] = `
<div>
  <div
    class="guided-answer"
  >
    <div
      class="guided-answer__header"
      style="padding-bottom: 20px;"
    >
      <div
        class="guided-answer__header__sub"
      >
        <div
          class="guided-answer__header__logoAndTitle"
        >
          <span
            id="sap-logo"
          >
            <svg />
          </span>
          <h1
            class="guided-answer__header__title"
          />
        </div>
      </div>
       
    </div>
    <main
      class="guided-answer__container"
      id="results-container"
    />
  </div>
</div>
`;

exports[`<App /> Match snapshot of component <App/> in loading mode 1`] = `
<div>
  <div
    class="guided-answer"
  >
    <div
      class="guided-answer__header"
      style="padding-bottom: 20px;"
    >
      <div
        class="guided-answer__header__sub"
      >
        <div
          class="guided-answer__header__logoAndTitle"
        >
          <span
            id="sap-logo"
          >
            <svg />
          </span>
          <h1
            class="guided-answer__header__title"
          />
        </div>
      </div>
       
    </div>
    <main
      class="guided-answer__container"
      id="results-container"
    >
      <div
        class="ms-Spinner root-110"
        id="loading-indicator"
      >
        <div
          class="ms-Spinner-circle ms-Spinner--large circle-111"
        />
      </div>
    </main>
  </div>
</div>
`;

exports[`<App /> Match snapshot of component <App/> with a guide selected 1`] = `
<div>
  <div
    class="guided-answer"
  >
    <div
      class="guided-answer__header"
      style="padding-bottom: 0px;"
    >
      <div
        class="ms-FocusZone css-114 guided-answer__header"
        data-focuszone-id="FocusZone1"
      >
        <div
          class="guided-answer__header__allAnswersButton"
        >
          <button
            class="guided-answer__header__navButtons home-icon"
            tabindex="0"
          >
            <svg
              class="guided-answer__header__navButtons__content"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M8.36 1.37l6.36 5.8-.71.71L13 6.964v6.526l-.5.5h-3l-.5-.5v-3.5H7v3.5l-.5.5h-3l-.5-.5V6.972L2 7.88l-.71-.71 6.35-5.8h.72zM4 6.063v6.927h2v-3.5l.5-.5h3l.5.5v3.5h2V6.057L8 2.43 4 6.063z"
                fill-rule="evenodd"
              />
            </svg>
             
            <span
              class="guided-answer__header__navButtons__content__text text-underline"
            />
          </button>
        </div>
        <div
          class="guided-answer__header__back-restart-buttons"
        >
          <button
            class="guided-answer__header__navButtons"
            tabindex="-1"
          >
            <svg
              class="guided-answer__header__navButtons__content"
              fill="currentColor"
              focusable="false"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M7 3.093l-5 5V8.8l5 5 .707-.707-4.146-4.147H14v-1H3.56L7.708 3.8 7 3.093z"
                fill-rule="evenodd"
              />
            </svg>
            <span
              class="guided-answer__header__navButtons__content text-underline"
              style="margin-left: 3px;"
            />
          </button>
          <div
            class="guided-answer__header__restart-feedback-buttons"
          >
            <button
              class="guided-answer__header__navButtons"
              tabindex="-1"
            >
              <i
                aria-hidden="true"
                class="ts-icon guided-answer__header__navButtons__content root-101"
                data-icon-name="Chat"
                style="padding-top: 2px;"
              >
                <svg
                  fill="none"
                  focusable="false"
                  viewBox="0 0 16 16"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 10H6V9H4.58578L3 10.5858V9H1V1H12V5H13V0H0V10H2V13L5 10Z"
                    fill="#C5C5C5"
                  />
                  <path
                    clip-rule="evenodd"
                    d="M16 6V13H14V16L11 13H7V6H16ZM13 12V13.5858L11.4142 12H8V7H15V12H13Z"
                    fill="#C5C5C5"
                    fill-rule="evenodd"
                  />
                </svg>
              </i>
            </button>
          </div>
        </div>
      </div>
    </div>
    <main
      class="guided-answer__container"
      id="results-container"
    >
      <section
        class="guided-answer__node__body"
      >
        <div
          class="column"
          id="left"
        >
          <nav
            class="container"
          >
            <div
              class="ms-FocusZone css-114"
              data-focuszone-id="FocusZone2"
              role="tree"
            >
              <div
                class="timeline-block"
                role="treeitem"
              >
                <button
                  class="timeline-content timeline-content-bottom-border"
                  tabindex="0"
                >
                  <div
                    class="timeline__path"
                    title="SAP Fiori Tools"
                  >
                    <span
                      class="timeline-content-title-small bold-text"
                    >
                      1
                    </span>
                    <span
                      class="timeline-content-title-large"
                    >
                      SAP Fiori Tools
                    </span>
                  </div>
                </button>
              </div>
            </div>
          </nav>
        </div>
        <div
          class="column"
          id="middle"
        >
          <div
            class="body_container"
          >
            <header>
              SAP Fiori Tools
            </header>
            <div
              id="hr"
            />
            <div
              class="ms-FocusZone css-114"
              data-focuszone-id="FocusZone3"
            >
              <div
                class="content"
              >
                <p>
                  SAP Fiori Tools is a set of extensions for SAP Business Application Studio and Visual Studio Code
                </p>
              </div>
            </div>
            <p
              class="guided-answer__node__question"
            >
              I have a problem with
            </p>
          </div>
          <div
            class="ms-FocusZone css-114"
            data-focuszone-id="FocusZone4"
            role="listbox"
          >
            <div
              class="guided-answer__node"
            >
              <button
                class="guided-answer__node__edge"
                role="option"
                tabindex="0"
              >
                Deployment
              </button>
              <button
                class="guided-answer__node__edge"
                role="option"
                tabindex="-1"
              >
                Fiori Generator
              </button>
            </div>
          </div>
        </div>
      </section>
    </main>
  </div>
</div>
`;

exports[`<App /> Match snapshot of component <App/> with results list 1`] = `
<div>
  <div
    class="guided-answer"
  >
    <div
      class="guided-answer__header"
      style="padding-bottom: 20px;"
    >
      <div
        class="guided-answer__header__sub"
      >
        <div
          class="guided-answer__header__logoAndTitle"
        >
          <span
            id="sap-logo"
          >
            <svg />
          </span>
          <h1
            class="guided-answer__header__title"
          />
        </div>
      </div>
       
    </div>
    <main
      class="guided-answer__container"
      id="results-container"
    >
      <div
        class="ms-FocusZone css-114"
        data-focuszone-id="FocusZone0"
      >
        <ul
          class="striped-list"
          role="listbox"
        >
          <div
            class="infinite-scroll-component__outerdiv"
          >
            <div
              class="infinite-scroll-component "
              style="height: auto; overflow: auto;"
            >
              <li
                class="tree-item"
                role="option"
              >
                <button
                  class="guided-answer__tree"
                  tabindex="0"
                >
                  <div
                    class="guided-answer__tree__ul"
                  >
                    <h3
                      class="guided-answer__tree__title"
                    >
                      SAP Fiori tools
                    </h3>
                    <div
                      class="bottom-section"
                    >
                      <span
                        class="guided-answer__tree__desc"
                      >
                        This is a troubleshooting guide to solve the issues while using SAP Fiori tools
                      </span>
                      <div
                        class="component-and-product-container"
                        style="margin-top: 10px;"
                      >
                        <div
                          class="guided-answer__tree__product"
                        >
                          <span
                            class="bottom-title"
                          >
                            Product: 
                          </span>
                          Product A
                        </div>
                        <div
                          class="guided-answer__tree__component"
                        >
                          <span
                            class="bottom-title"
                          >
                            Component: 
                          </span>
                          comp-a
                        </div>
                      </div>
                    </div>
                  </div>
                </button>
              </li>
            </div>
          </div>
        </ul>
      </div>
    </main>
  </div>
</div>
`;
